providers/yttcb/clusterbootstrap.yaml

```diff
@@ -315,7 +362,11 @@ spec:
   additionalPackages:
     - refName: metrics-server*
     - refName: secretgen-controller*
+    #@ if pinniped_configs_exist():
     - refName: pinniped*
+      valuesFrom:
+        secretRef: #@ "{}-pinniped-addon".format(data.values.CLUSTER_NAME)
```

Comment:
We tried creating a `management-cluster` using:

```bash
DEPLOY_TKG_ON_VSPHERE7=true tanzu management-cluster create -f tkg-mgmt-vc-with-pinniped.yaml --plan dev
```
with `tkg-mgmt-vc-with-pinniped.yaml` being a basic config with some pinniped env vars:

```yaml
# default mgmt config vals:
CLUSTER_NAME: tkg-mgmt-vc-with-pinniped
CLUSTER_PLAN: dev
INFRASTRUCTURE_PROVIDER: vsphere
... trimmed for brevity

# pinniped config
# --------------------
IDENTITY_MANAGEMENT_TYPE: ldap

# LDAP
# using VMware LDAP config:
# LDAP_BIND_DN:                                                 # skip this field
# LDAP_BIND_PASSWORD:                                           # skip this field
LDAP_HOST: ldaps.eng.vmware.com:636
LDAP_USER_SEARCH_BASE_DN: 'ou=people,dc=vmware,dc=com'
LDAP_USER_SEARCH_FILTER: '(objectClass=posixAccount)'
LDAP_USER_SEARCH_USERNAME: uid            # i guess?
LDAP_USER_SEARCH_ID_ATTRIBUTE: uid
LDAP_USER_SEARCH_EMAIL_ATTRIBUTE: uid
LDAP_USER_SEARCH_NAME_ATTRIBUTE: uid
LDAP_GROUP_SEARCH_BASE_DN: dc=vmware,dc=com
LDAP_GROUP_SEARCH_FILTER: (objectClass=posixGroup)
LDAP_GROUP_SEARCH_USER_ATTRIBUTE: uid
LDAP_GROUP_SEARCH_GROUP_ATTRIBUTE: memberUid
LDAP_GROUP_SEARCH_NAME_ATTRIBUTE: cn
# LDAP_ROOT_CA_DATA_B64:                                        # not sure2

```

which created the expected `pinniped secret` as well as this cluster bootstrap file:

```yaml
# cat .config/tanzu/tkg/clusterconfigs/tkg-mgmt-vc-with-pinniped.yaml
# generated when we ran tanzu cluster create above...
...
apiVersion: run.tanzu.vmware.com/v1alpha3
kind: ClusterBootstrap
metadata:
  annotations:
    tkg.tanzu.vmware.com/add-missing-fields-from-tkr: v1.23.8---vmware.2-tkg.2-zshippable
  name: tkg-mgmt-vc-with-pinniped
  namespace: tkg-system
spec:
  additionalPackages:
  - refName: metrics-server*
  - refName: secretgen-controller*
  # this is what we expressed in our template logic.
  # the *-pinniped-addon is the correct name.
  # but does not match in the next block below
  - refName: pinniped*
    valuesFrom:
      secretRef: tkg-mgmt-vc-with-pinniped-pinniped-addon  # pinniped-addon, not pinniped-package!
...
```
However, the actual cluster boostrap file on the cluster does not look like the configuration file that we generated and then used to create the cluster.  It has defaulted back to the default pinniped package secret:

```yaml
# kubectl get clusterbootstrap tkg-mgmt-vc-with-pinniped -n tkg-system -o yaml
apiVersion: run.tanzu.vmware.com/v1alpha3
kind: ClusterBootstrap
metadata:
  creationTimestamp: "2022-11-16T19:59:20Z"
  generation: 1
  name: tkg-mgmt-vc-with-pinniped
  namespace: tkg-system
  ownerReferences:
  - apiVersion: cluster.x-k8s.io/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: Cluster
    name: tkg-mgmt-vc-with-pinniped
    uid: 91a8faba-5b3b-47cb-a047-4ed88cb49a13
  resourceVersion: "6456"
  uid: 3f7daf58-131a-4e63-b649-f8f2b03ee776
spec:
  additionalPackages:
  - refName: metrics-server.tanzu.vmware.com.0.6.1+vmware.1-tkg.3-zshippable
  - refName: secretgen-controller.tanzu.vmware.com.0.11.0+vmware.2-tkg.1-zshippable
  - refName: tkg-storageclass.tanzu.vmware.com.0.28.0-dev-115-gd7751adc+vmware.1
  - refName: pinniped.tanzu.vmware.com.0.12.1+vmware.2-tkg.3-zshippable
   # this is for the default valuesFrom,
   # not the updated valuesFrom *-pinniped-addon that we 
   # changed.  Something is incorrect here.
    valuesFrom:
      secretRef: tkg-mgmt-vc-with-pinniped-pinniped-package # pinniped-package? this is wrong.
...
```
However, it did create the `pinniped-addon secret`, as we expected it to:

```bash
kubectl get secret tkg-mgmt-vc-with-pinniped-pinniped-addon -n tkg-system
# NAME                                       TYPE                         DATA   AGE
# tkg-mgmt-vc-with-pinniped-pinniped-addon   tkg.tanzu.vmware.com/addon   1      11m
```

It appears that we declared what we want, but something overrode our declaration with the default?

Reply:
This was resolved by https://github.com/vmware-tanzu/tanzu-framework/pull/3960.